cleanlab/internal/multiannotator_utils.py

```diff
@@ -298,18 +299,32 @@ def find_best_temp_scaler(
     and model pred_probs"""
 
     soft_cross_entropy_coarse = np.full(len(coarse_search_range), np.NaN)
+    log_pred_probs = np.log(
+        pred_probs, where=pred_probs > 0, out=np.full(pred_probs.shape, -np.inf)
+    )
     for i, curr_temp in enumerate(coarse_search_range):
-        # clip pred_probs to prevent taking log of 0
-        pred_probs = np.clip(pred_probs, a_min=SMALL_CONST, a_max=None)
-        pred_probs = pred_probs / np.sum(pred_probs, axis=1)[:, np.newaxis]
-
-        log_pred_probs = np.log(pred_probs) / curr_temp
-        scaled_pred_probs = np.exp(log_pred_probs) / np.sum(np.exp(log_pred_probs))  # softmax
+        scaled_pred_probs = softmax(log_pred_probs, temperature=curr_temp, axis=1, shift=False)
         soft_cross_entropy_coarse[i] = np.mean(
             compute_soft_cross_entropy(labels_multiannotator, scaled_pred_probs)
         )
 
     min_entropy_ind = np.argmin(soft_cross_entropy_coarse)
+    fine_search_range = _set_fine_search_range(
+        coarse_search_range, fine_search_size, min_entropy_ind
+    )
+    soft_cross_entropy_fine = np.full(len(fine_search_range), np.NaN)
+    for i, curr_temp in enumerate(fine_search_range):
+        scaled_pred_probs = softmax(log_pred_probs, temperature=curr_temp, axis=1, shift=False)
+        soft_cross_entropy_fine[i] = np.mean(
+            compute_soft_cross_entropy(labels_multiannotator, scaled_pred_probs)
+        )
+    best_temp = fine_search_range[np.argmin(soft_cross_entropy_fine)]
+    return best_temp
+
+
+def _set_fine_search_range(
+    coarse_search_range: list, fine_search_size: int, min_entropy_ind: int
+) -> np.ndarray:
```

Comment:
```suggestion
) -> int:
```

this will probably fix the failing type check